modules/cloudbuild/main.tf

```diff
@@ -108,6 +109,16 @@ resource "time_sleep" "impersonate_propagation" {
   ]
 }
 
+/******************************************
+  Cloudbuild Service Agent Role
+******************************************/
+
+resource "google_project_iam_member" "cb_service_agent_role" {
```

Comment:
This should be fine, but should be automatically granted: https://cloud.google.com/build/docs/securing-builds/configure-access-for-cloud-build-service-account#service-agent-permissions

I wonder if the propagation time is causing an issue with the CI?